### PR TITLE
Refactored code and fixed the profile-show component

### PR DESCRIPTION
**Pull request by DCox2016**

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <header></header>
-<profile-search></profile-search>
+<profile-show></profile-show>
```
Comment (cpyle0819) [inline src/app/app.component.html]: Try to work for consistent naming of your components. Here you have `profile-show`, but below you are prefixed with 'app'. Angular's styleguide recommends having an app prefix. `app-profile-show` would be fine for example.  If you have an actual name or abbreviation for your app, that would be better. So if your app is 'GitHub Viewer', `ghv-profile-show` would be fine too.

File: src/app/profile-search/profile-search.service.ts [hunk 2]
```diff
@@ -1,52 +1,39 @@
...
+  profile: BehaviorSubject<ProfileInfo> = new BehaviorSubject({ login: '', name: '', avatar_url: '', repos_url: '' });
+  repos: BehaviorSubject<any[]> = new BehaviorSubject([]);
+ 
+ 
+  constructor(private http: HttpClient) {
+    this.profile.subscribe(({ repos_url }) => {
+      if (repos_url) {
+        // http request, set repoFetch to return value
+        this.http.get(repos_url).pipe(
+          tap(repos => this.repos.next(repos as any[]))
```
Comment (cpyle0819) [inline src/app/profile-search/profile-search.service.ts]: You can remove the `tap` here, and just do `this.http.get(repos_url).subscribe(this.repos)`

File: src/app/profile-show/profile-show.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  providers: [ ProfileSearchService ],
+  styleUrls: ['./profile-show.css']
+})
+
+
+export class ShowProfileComponent implements OnInit {
+  @ViewChild('searchInput') searchInput: ElementRef;
+  user$: Observable<any>;
+  repos: Observable<any[]>;
+  user = this.searchService.profile;
```
Comment (cpyle0819) [inline src/app/profile-show/profile-show.component.ts]: I don't think this is needed anymore

File: src/app/profile-search/profile-search.service.ts [hunk 4]
```diff
@@ -1,52 +1,39 @@
...
+    });
   }
 
-  search (packageName: string, refresh = false): Observable<PackageInfo>
-   {
-    const options = createHttpOptions(packageName, refresh);
-    this.searchResults = this.http.get(searchUrl + packageName, options) as Observable<PackageInfo>;
-    return this.searchResults
+  search (packageName: string): Observable<ProfileInfo> {
+    const searchReturn = this.http.get(searchUrl + packageName) as Observable<ProfileInfo>;
```
Comment (cpyle0819) [inline src/app/profile-search/profile-search.service.ts]: You can return `this.http.get..` directly. You do not have to declare `searchReturn` and return it.
```suggestion
    return this.http.get(searchUrl + packageName) as Observable<ProfileInfo>;
```

File: src/app/profile-show/profile-show.component.ts (hunk 3 above)
Comment (cpyle0819) [inline src/app/profile-show/profile-show.component.ts]: I don't think you need this. Your user info is stored in `user$`.

File: src/app/profile-show/profile-show.component.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  templateUrl: './profile-show.html',
+  providers: [ ProfileSearchService ],
+  styleUrls: ['./profile-show.css']
+})
+
+
+export class ShowProfileComponent implements OnInit {
+  @ViewChild('searchInput') searchInput: ElementRef;
+  user$: Observable<any>;
+  repos: Observable<any[]>;
```
Comment (cpyle0819) [inline src/app/profile-show/profile-show.component.ts]: This should be be set to `this.searchService.repos`

File: src/app/profile-search/profile-search.service.ts [hunk 6]
```diff
@@ -1,52 +1,39 @@
...
-    httpErrorHandler: HttpErrorHandler
-   ) {
-    this.handleError = httpErrorHandler.createHandleError('Service');
+export class ProfileSearchService {
+  profile: BehaviorSubject<ProfileInfo> = new BehaviorSubject({ login: '', name: '', avatar_url: '', repos_url: '' });
+  repos: BehaviorSubject<any[]> = new BehaviorSubject([]);
+ 
+ 
+  constructor(private http: HttpClient) {
+    this.profile.subscribe(({ repos_url }) => {
```
Comment (cpyle0819) [inline src/app/profile-search/profile-search.service.ts]: **BROKEN**
This is the area I mentioned in Slack. You are subscribing to 'profile' but there is nothing that will cause this `profile` to emit anything. So the call for repos will never happen.

File: src/app/ObsInterface.ts [hunk 7]
```diff
@@ -4,6 +4,3 @@ export class RepoList {
     stargazers_count: number;
     language: string;
 }
```
[the remark this answers is not in the document]
Comment (DCox2016) [reply src/app/ObsInterface.ts]: This was fixed, but must of changed back when stashed my changes

**Comment by DCox2016** [comment]
cleaned up the existing code and fixed the profile show component

**Pull request merged**